### ISSUE TITLE
Unpredictability with sites such as GitHub

### ISSUE THREAD

**Issue by grahamperrin** [Ulf3000/sessionBuddy#3]
Prior to a save, note the comment beneath the opening post: 

![2018-10-29 06 25 03 waterfox](https://user-images.githubusercontent.com/192271/47633966-c0e20900-db47-11e8-8cdf-409bfd7a1d24.png)
 
Some time after a restoration (probably **not** the subsequent restoration), note the absence of the comment: 

![2018-10-29 06 47 42 waterfox](https://user-images.githubusercontent.com/192271/47634008-e8d16c80-db47-11e8-8aaf-57f26f1b48b0.png)

Unfortunately I did not capture the steps in between, but I almost certainly used [Tab Mover](https://addons.mozilla.org/addon/tab-mover/) for the movement between windows. 

Can you think of any sane explanation? 

Might the staleness occur if a quit occurs too soon after a SessionBuddy save? Or (given my wild array of extensions) might something at quit time cause corruption or loss of saved data? 

### Environment

```text
$ date ; uname -v
Mon 29 Oct 2018 07:12:26 GMT
FreeBSD 13.0-CURRENT r339737 GENERIC-NODEBUG 
$ pkg query '%o %v %R' waterfox
www/waterfox 56.2.3_2 poudriere
$ 
```

### Generally (not specific to SessionBuddy)

The staleness is _very_ often a problem with sites such as GitHub. 

If we can't find a reasonable explanation, then I should probably continue to treat the session storage in Waterfox 56.x as inherently (Firefox 56.0.2) unreliable. 

### Related

Mozilla bug [1330633 - (ss-reliability) Sessionstore reliability tracking](https://bugzilla.mozilla.org/show_bug.cgi?id=1330633)

[workaround for staleness (outdated content) when e.g. Firefox is used to view GitHub issues · Issue #672 · sindresorhus/refined-github](https://github.com/sindresorhus/refined-github/issues/672) (2017-08-17) reminds me that for the problems with GitHub, I was clutching at straws long before I discovered Waterfox. There's reference to: 

- Mozilla bug [706970 - Tabs restored at startup (app tabs, session restore) should be refreshed from server, not in a "stale" state from cache](https://bugzilla.mozilla.org/show_bug.cgi?id=706970)

– however, later in the issue, with added emphasis: 

> tl;dr re the conversation that followed <http://logs.glob.uno/?c=mozilla%23firefox&s=13+Sep+2017&e=13+Sep+2017#c450899> I no longer suspect a cache-related problem. More likely **a bug with, or affecting, session storage**.

**Comment by Ulf3000** [comment]
nahh thats just because i havent put an unloader coder into the addon yet, means when it shuts down it doesnt destroy the menu but generates a new one when enabled so you end up with 2 ,3 ,4 menus ..

leave this open please .. for now i think its alright since it doesnt harm  but its obviously not optimised , i was just too lazy at that point to write the unloader, treat it as a kind of script right now then a full fledged extension.. but ill do it eventually

**Comment by grahamperrin** [comment]
OK, I'll close this issue. Without using SessionBuddy I coincidentally reproduced what seems to be a session storage issue whilst using Discord. 

> OK I'm going to abandon Discord but it was coincidentally a good opportunity to demonstrate what is, I think, seriously messed up session storage -- the 'Now you see it, now you don't' effect after closing then reopening a tab.
